source/yats/include/yats/pipeline.h

```diff
@@ -16,14 +16,14 @@ class pipeline
 public:
     pipeline() = default;
 
-    template <typename Task>
-    task_configurator<Task>* add()
+    template <typename Task, typename... Parameters>
+    task_configurator<Task, Parameters...>* add(Parameters&&... parameters)
     {
         static_assert(has_unique_ids_v<typename decltype(make_helper(&Task::run))::wrapped_input>, "Can not add Task because multiple Inputs share the same Id.");
         static_assert(has_unique_ids_v<typename decltype(make_helper(&Task::run))::return_base>, "Can not add Task because multiple Outputs share the same Id.");
 
-        m_tasks.push_back(std::make_unique<task_configurator<Task>>());
-        return static_cast<task_configurator<Task>*>(m_tasks.back().get());
+        m_tasks.push_back(std::make_unique<task_configurator<Task, Parameters...>>(std::forward<Parameters>(parameters)...));
```

Comment:
It currently looks like that the tuple holds references. This is not a good idea because we can not control the lifetime of them, so we should always store the parameters as values. The STL behaves the same way when storing parameters of a function that will be called at a later time (bind, thread). This can be implemented by removing the reference qualifier from the type and forwarding the parameter pack into the construction of the tuple.

When kepping the value is important the user can always use pointers instead or use std::ref or std::cref.